crates/store/re_chunk_store/src/query.rs

```diff
@@ -75,6 +118,110 @@ impl ChunkStore {
             .map_or(false, |components| components.contains(component_name))
     }
 
+    /// Check whether an entity has a specific component on any timeline, or in its static data.
+    pub fn entity_has_component_on_any_timeline(
+        &self,
+        entity_path: &EntityPath,
+        component_name: &ComponentName,
+    ) -> bool {
+        re_tracing::profile_function!();
+
+        if self
+            .static_chunk_ids_per_entity
+            .get(entity_path)
+            .and_then(|static_chunks_per_component| static_chunks_per_component.get(component_name))
+            .and_then(|id| self.chunks_per_chunk_id.get(id))
+            .is_some()
+        {
+            return true;
+        }
+
+        for temporal_chunk_ids_per_component in self
+            .temporal_chunk_ids_per_entity_per_component
+            .get(entity_path)
+            .iter()
+            .flat_map(|temporal_chunk_ids_per_timeline| temporal_chunk_ids_per_timeline.values())
+        {
+            if temporal_chunk_ids_per_component
+                .get(component_name)
+                .is_some()
+            {
+                return true;
+            }
+        }
+
+        false
+    }
```

Comment:
Okay, now that I'm familiar with the entire PR, I can tell what's going on.

See my comment further down:
> Okay I'm starting to see where the ambiguities stem from.
> 
> What do you really want to know here: whether the entity has some component associated with it (which, once true, will always be true -- no matter of what the GC ends up removing), or whether the entity has some events available in the store (which can be true at some point but then false at a later time because GC)?
> 
> This method should make it clear which one of these is at play here.
> 
> We need to make that clear all over the place in all these store APIs and their docs, in fact.

Every store API must make it very clear which camp it falls into: is it about checking whether a relationship exists between an entity and a component (which is intemporal -- once true, it will always stay that way), or is it about counting the number of events currently in storage (which can be non-zero a some time `t` and then back to zero at a later time `t+n`).